src/test/routes/safe/store/actions/utils.test.ts

```diff
@@ -0,0 +1,170 @@
+
+    // when
+    const result = await shouldExecuteTransaction(safeInstance, nonce, lastTx)
+
+    // then
+    expect(result).toBe(false)
+  })
+  it('It should return true if given a safe with a threshold === 1 and the last transaction is first transaction (nonce 0)',  async () => {
+    // given
+    const nonce = '0'
+    const threshold = '1'
+    const safeInstance = getMockedSafeInstance({ threshold })
+    const lastTx = getMockedTxServiceModel({})
+
+    // when
+    const result = await shouldExecuteTransaction(safeInstance, nonce, lastTx)
+
+    // then
+    expect(result).toBe(true)
+  })
+  it('It should return true if given a safe with a threshold === 1 and the last transaction is not the first transaction, the last transaction is already executed',  async () => {
+    // given
+    const nonce = '10'
+    const threshold = '1'
+    const safeInstance = getMockedSafeInstance({ threshold })
+    const lastTx = getMockedTxServiceModel({ isExecuted: true })
+
+    // when
+    const result = await shouldExecuteTransaction(safeInstance, nonce, lastTx)
+
+    // then
+    expect(result).toBe(true)
+  })
+  it('It should return false if given a safe with a threshold === 1 and the last transaction is not first transaction, the last transaction is not already executed',  async () => {
+    // given
+    const nonce = '10'
+    const threshold = '1'
+    const safeInstance = getMockedSafeInstance({ threshold })
+    const lastTx = getMockedTxServiceModel({ isExecuted: false })
+
+    // when
+    const result = await shouldExecuteTransaction(safeInstance, nonce, lastTx)
+
+    // then
+    expect(result).toBe(false)
+  })
+})
+
+describe('getNewTxNonce', () => {
+  it('It should return 2 if given the last transaction with nonce 1',  async () => {
+    // given
+    const safeInstance = getMockedSafeInstance({})
+    const lastTx = getMockedTxServiceModel({ nonce: 1 })
+    const expectedResult = '2'
+
+    // when
+    const result = await getNewTxNonce(undefined, lastTx, safeInstance)
+
+    // then
+    expect(result).toBe(expectedResult)
+  })
+  it('It should return 0 if given a safe with the last nonce 0 and no lastTransaction, also should call safeInstance nonce method',  async () => {
+    // given
+    const safeNonce = '0'
+    const safeInstance = getMockedSafeInstance({ nonce: safeNonce})
+    const expectedResult = '0'
+    const mockFnCall = jest.fn().mockImplementation(() => safeNonce)
+    const mockFnNonce = jest.fn().mockImplementation(() => ({call: mockFnCall}))
+
+    safeInstance.methods.nonce = mockFnNonce
+
+    // when
+    const result = await getNewTxNonce(undefined, null, safeInstance)
+
+    // then
+    expect(result).toBe(expectedResult)
+    expect(mockFnNonce).toHaveBeenCalled()
+    expect(mockFnCall).toHaveBeenCalled()
+    mockFnNonce.mockRestore();
+    mockFnCall.mockRestore();
+  })
+  it('Given a Safe and a last transaction, should return the lastTransactionNonce + 1',  async () => {
```

Comment:
`Given a Safe and the last transaction, should return nonce of the last transaction + 1`